src/JsParsing/BasicJsExpressionParser.php

```diff
@@ -11,10 +11,10 @@ class BasicJsExpressionParser implements JsExpressionParser {
 	 */
 	public function parse( $expression ) {
 		$expression = $this->normalizeExpression( $expression );
-		if ( strpos( $expression, '!' ) === 0 ) { // ! operator application
+		if ( strncmp( $expression, '!', 1 ) === 0 ) {
```

Comment:
Oh, by the way, an other perfectly valid and equally fast possibility is `substr( $expression, 0, 1 ) === '!'`. Just recently we discussed this in the MediaWiki CodeSniffer project, and preferred substr over strncmp, which might be a little obscure.